src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsauditapi/jpa/AuditRepository.kt

```diff
@@ -1,9 +1,23 @@
 package uk.gov.justice.digital.hmpps.hmppsauditapi.jpa
 
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.Pageable
+import org.springframework.data.jpa.repository.Query
 import org.springframework.data.repository.PagingAndSortingRepository
 import org.springframework.stereotype.Repository
 import uk.gov.justice.digital.hmpps.hmppsauditapi.listeners.HMPPSAuditListener.AuditEvent
 import java.util.UUID
 
 @Repository
-interface AuditRepository : PagingAndSortingRepository<AuditEvent, UUID>
+interface AuditRepository : PagingAndSortingRepository<AuditEvent, UUID> {
+  @Query(
+    """
+    select 
+        ae from AuditEvent ae 
+    where 
+         (coalesce(:what) is null or ae.what=:what)
+     and (coalesce(:who) is null or ae.who=:who)
+    """
+  )
+  fun findPage(pageable: Pageable, who: String?, what: String?): Page<AuditEvent>
```

Comment:
I didn't realise that `@Param` isn't needed if the parameter name is the same. Nice.

Reply:
yes - magic!
Annoyingly tho - I had to do a coalesce(:who) rather than just :who is null - the latter worked fine in tests(h2 db) but not directly in the postgres db

Reply:
That's interesting - I only had to do that for a list.